### DISCUSSION TITLE
Passing additional user selected information as part of raising an intent

### DISCUSSION THREAD

**Discussion by hampshan** [finos/FDC3#572] [Ideas]
I have a use case where we'd like to be able to raise intents but with additional information to pass to our "layout manager".

For example, we may have a right-click context menu that presents the user with a variety of intents.  These intents could be handled by the layout manager in different ways, e.g. open as a new window,  or open in an existing collection of apps, etc.

To date the FDC3 spec does not have a way to pass a preferred layout choice (or anything else that is unrelated to the context or TargetApp).

Should TargetApp be enhanced to include user preferences selected at the same time as raising the intent?

**Comment by kriswest** [comment]
Of the options you give as an example:
> open as a new window, or open in an existing collection of apps, etc.

The first is the default behaviour (sans any layout info), while the later might be taken care of via the (recently approved) targeting of app instances by intents (by passing an `AppMetadata` object which includes an `instanceId` as the `target` parameter):
- #509 

which would allow you to target an instance you already have details for (e.g. because you opened it or it previously resolved an intent for you) or that you have retrieved details of via `fdc3.findIntent()` or the new `fdc3.findInstances()`. As I mentioned at last week's meeting, several FDC3 1.2 implementations already allow you to select an existing app instance through a resolver UI, but this is not yet possible via the API until the above proposal is released as part of FDC3 2.0. Note #509 also includes an `instanceMetadata` element allowing the passing of additional metadata to help identify an instance, however, its content is not yet standardized.

I would assume the next extension of your query is passing layout information (dimensions, position, monitor) for a new instance of the app. Whilst I think this is entirely logical, FDC3 has previously avoided anything to do with 'window management'. Hence, if you think it definitely is needed I would recommend outlining the use-case and an example of use that we can discuss at an upcoming meeting.

**Reply by kriswest** [reply]
Hi @hampshan does being able to target existing app instances sufficiently solve for your use case or do we need to add this issue to a future meeting for discussion? If so can you provide some more detail on your use-case and support you are looking for in the API?
